Various Fixes, catches.

Datraen opened the pull request:
Fixes #16666 by adding a dropInto method of removal for ID cards when a silicon user removes an ID card from the uniform vendor.

Catches cases that cause #16677 by checking to make sure that the turf is valid. I'm not entirely sure what caused the runtime, but the target human may have been qdel'd while processing.

Tentatively fixes #16674 by checking the returned value of confirm(), returning PROCESS_KILL. Grabs are qdel'd in confirm().

Datraen committed:
Fixes #16666

PsiOmegaDelta commented on maps/torch/items/uniform_vendor.dm:
```diff
@@ -56,9 +56,14 @@
+			else
+				ID.dropInto(loc)
+				ID = null
+				selected_outfit.Cut()
```
Seems to me there was no need to place the last two ID/selected_outfit.Cut() lines within the if/else check.

Datraen replied:
Corrected, I don't know why I did that.

Datraen committed:
Tentatively fixes #16674

Datraen committed:
Catches #16677

Cases where the mob no longer exist would result in a null loc, unsure of direct cause.
